lib/iris/fileformats/grib/__init__.py

```diff
@@ -692,10 +692,6 @@ def _message_values(grib_message, shape):
     data = gribapi.grib_get_double_array(grib_message, 'values')
     data = data.reshape(shape)
 
-    # Handle missing values in a sensible way.
-    mask = np.isnan(data)
-    if mask.any():
-        data = ma.array(data, mask=mask, fill_value=np.nan)
```

Comment:
@lbdreyer Yup, keep it all `nan`s and deal with it later, rather than creating the masked array here. Got it.